### PR TITLE
Add migration docs

### LINKED ISSUE #2014: migration 2.4->3.1.3 issue related to EnumerationGraphType
In 2.4. i wanted to enrich some enums with additional fields and had a graphtype which had an Enum as datasource and the same enum value as EnumerationGraphType<T> as field. This worked in 2.4. but in 3.1.3 it throws an InvalidOperationException.

Is this not supposed to be used like this anymore or is it a bug?

image

image

### PR DESCRIPTION
**Pull request by sungam3r**
fixes #2014

### CODE REVIEW

File: docs2/site/docs/migrations/migration3.md [hunk 1]
```diff
@@ -47,6 +47,23 @@ This project now requires the .NET Standard 2.0 and breaks compatibility with ap
...
+Examples:
+
+| .NET GraphType Name | Default Name |
+|---------------------|--------------|
+| MoneyType           | MoneyType    |
+| MoneyGraphType      | Money        |
+
+You can always set your own type name either in the constructor or externally. The result of this change is that it is no
+longer possible to have Graph Types with unspecified (`null`) or invalid names. This allows to avoid a number of errors
+and incomprehensible behavior at runtime.
```
Comment (Shane32) [inline docs2/site/docs/migrations/migration3.md:65]:
```suggestion
You can always set your own type name either in the constructor or externally.
```
I deleted some text that seems irrelevant.  Whatever you want is fine, however.

**Review by Shane32 — COMMENTED** [review]
This isn't a breaking change, is it?  All 2.4.x code, which required the name be set in the graph type constructor, should work properly in 3.x.  So I would think this belongs in the FAQ.

**Comment by sungam3r** [comment]
It seems to be a breaking change. Names in version 2.4.0 were set to different values but I don't remember exactly. @dBug404 ?

File: docs2/site/docs/migrations/migration3.md (hunk 1 above)
Comment (sungam3r) [reply docs2/site/docs/migrations/migration3.md:65]: I would like to leave a mention of the motivation behind this decision.

**Comment by dBug404** [comment]
It is a breaking change. After the upgrade I have to modify all mutation parameter names which have a class name with suffix GraphType either on the server or the client side.
The client mutation calls include the names of the graphtype for the parameters:
image

File: src/GraphQL.sln [hunk 2]
```diff
@@ -73,7 +73,6 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "guides", "guides", "{F279F2
 		..\docs2\site\docs\guides\known-issues.md = ..\docs2\site\docs\guides\known-issues.md
 		..\docs2\site\docs\guides\links.md = ..\docs2\site\docs\guides\links.md
 		..\docs2\site\docs\guides\migration.md = ..\docs2\site\docs\guides\migration.md
```
Comment (Shane32) [inline src/GraphQL.sln:75]:
```suggestion
```

**Review by Shane32 — APPROVED** [review]

File: src/GraphQL.sln (hunk 2 above)
Comment (Shane32) [reply src/GraphQL.sln:75]: This file doesn't exist.
